app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java

```diff
@@ -1,79 +1,142 @@
     }
 
     @Override
-    public String getFullResult() {
-
-        switch (operator) {
-
-            case OPERATOR_SUM:
-                if (result == EMPTY_STRING) {
-                    result = String.valueOf(Double.parseDouble(firstOperand) + Double.parseDouble(secondOperand));
-                }
-
-                break;
-
-            case OPERATOR_MINUS:
-                result = String.valueOf(Double.parseDouble(firstOperand) - Double.parseDouble(secondOperand));
-                break;
-
-            case OPERATOR_DIVIDE:
-                result = String.valueOf(Double.parseDouble(firstOperand) / Double.parseDouble(secondOperand));
-                break;
+    public String getPartialResult() {
+        return getValue();
+    }
 
-            case OPERATOR_MULTIPLY:
-                result = String.valueOf(Double.parseDouble(firstOperand) * Double.parseDouble(secondOperand));
-                break;
+    private boolean isValidOperation() {
+        if (operator.isEmpty()) {
+            if (firstOperand.isEmpty()) {
+                result = EMPTY_STRING;
+            } else {
+                result = String.valueOf(firstOperand.getValue());
+            }
+            return false;
+        } else if (secondOperand.isEmpty()) {
+            result = EMPTY_STRING;
+            error = ErrorUtils.ERROR_MESSAGE_INVALID_FORMAT;
+            return false;
+        }
+        return true;
+    }
 
-            default:
-                result = ERROR_MESSAGE;
+    @Override
+    public void doOperations() {
+
+        if (isValidOperation()) {
+
+            switch (operator) {
+
+                case OPERATOR_SUM:
+                    result = String.valueOf(firstOperand.getValue() + secondOperand.getValue());
+                    break;
+
+                case OPERATOR_MINUS:
+                    result = String.valueOf(firstOperand.getValue() - secondOperand.getValue());
+                    break;
+
+                case OPERATOR_DIVIDE:
+                    if (secondOperand.getValue() == ZERO_NUMBER_DOUBLE_TYPE) {
+                        result = EMPTY_STRING;
+                        error = ErrorUtils.ERROR_MESSAGE_DIVISION;
+                    } else {
+                        result = String.valueOf(firstOperand.getValue() / secondOperand.getValue());
+                    }
+                    break;
+
+                case OPERATOR_MULTIPLY:
+                    result = String.valueOf(firstOperand.getValue() * secondOperand.getValue());
+                    break;
+
+                default:
+                    result = EMPTY_STRING;
+                    error = ErrorUtils.ERROR_MESSAGE;
+                    break;
+            }
+            checkNumberSign();
+
+            error = ErrorUtils.NONE;
+            secondOperand.eraseOperands();
         }
+    }
 
+    public String getResult() {
         return result;
     }
 
+    private void checkNumberSign() {
```

Comment:
the name don't describe the function, thisshould be updateFirstOperand

Reply:
Done